CompanionPane/src/androidTest/java/com/microsoft/device/display/samples/companionpane/LayoutTest.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.device.display.samples.companionpane
+
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.window.testing.layout.WindowLayoutInfoPublisherRule
+import com.microsoft.device.display.samples.companionpane.ui.theme.CompanionPaneAppTheme
+import com.microsoft.device.dualscreen.testutils.getString
+import com.microsoft.device.dualscreen.testutils.simulateHorizontalFold
+import com.microsoft.device.dualscreen.testutils.simulateVerticalFold
+import com.microsoft.device.dualscreen.windowstate.rememberWindowState
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.rules.TestRule
+
+class LayoutTest {
+    private val composeTestRule = createAndroidComposeRule<MainActivity>()
+    private val publisherRule = WindowLayoutInfoPublisherRule()
+    private val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    @get: Rule
+    val testRule: TestRule
+
+    init {
+        testRule = RuleChain.outerRule(publisherRule).around(composeTestRule)
+        RuleChain.outerRule(composeTestRule)
+    }
+
+    /**
+     * Tests that the single portrait layout appears when one pane is shown and the device is in
+     * the portrait orientation
+     */
+    @Test
+    fun app_testSinglePortraitLayout() {
+        composeTestRule.setContent {
+            CompanionPaneAppTheme {
+                CompanionPaneApp(composeTestRule.activity.rememberWindowState())
+            }
+        }
+
+        // Lock in portrait orientation
+        device.setOrientationNatural()
+
+        // Check that single portrait layout is shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.single_port))
+            .assertIsDisplayed()
+
+        // Unfreeze orientation
+        device.unfreezeRotation()
```

Comment:
Since we removed `freeRotation`, do we still need `unfreezeRotation`?

Reply:
Yes, but good catch! It turns out that `freezeRotation` was redundant because `setOrientation` already calls `freezeRotation`.

[![screenshot of android developer docs for UiDevice method setOrientationLeft that shows how freezeRotation is already called within the method](https://user-images.githubusercontent.com/33138268/146830016-63e7ed01-0f22-4edb-989e-00ce0c558e01.png)](https://developer.android.com/reference/androidx/test/uiautomator/UiDevice#setorientationleft)